### ISSUE TITLE
testing code does not work, the results are not correct, mAP values are easy 0.125, medium 0.134, hard 0.156  

### ISSUE THREAD

**Issue by hdjsjyl** [deepinsight/mxnet-SSH#8]
Both model I trained by this code or your pretrained model doesnot work for the testing code. And I saw your SSH folder in the insightface project. The testing code of that folder detects faces via original image, I do not think, it can achieve 81.4% of mAP for hard set of validation set of WIDER FACE. So could you provide some advice? Thanks

**Comment by hdjsjyl** [comment]
@nttstar  when I change the function 'generate_anchors_fpn' to this:
def generate_anchors_fpn(base_size, scales, ratios):
    anchors = []
    _ratios = ratios.reshape((len(base_size), -1))
    _scales = scales.reshape((len(base_size), -1))
    for i, bs in enumerate(base_size):
        __ratios = _ratios[i]
        __scales = _scales[i]
        # print('anchors_fpn', bs, __ratios, __scales, file=sys.stderr)
        r = generate_anchors(bs, __ratios, __scales)
        # print('anchors_fpn', r.shape, file=sys.stderr)
        anchors.append(r)

    return  anchors

It works. But The results of your pretrained model with single scale(1200, 1600) testing, the mAP value for hard set of validation set of WIDER FACE only achieves 78.2. I do not know the reason, do you have some suggestions? thank you.
